src/Symfony/Component/Notifier/Bridge/TurboSms/TurboSmsTransport.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\TurboSms;
+
+use Symfony\Component\Notifier\Exception\LengthException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Artem Henvald <genvaldartem@gmail.com>
+ *
+ * @see https://turbosms.ua/api.html
+ */
+final class TurboSmsTransport extends AbstractTransport
+{
+    protected const HOST = 'api.turbosms.ua';
+
+    private const SUBJECT_LATIN_LIMIT = 1521;
+    private const SUBJECT_CYRILLIC_LIMIT = 661;
+    private const SENDER_LIMIT = 20;
+
+    private $authToken;
+    private $from;
+
+    public function __construct(string $authToken, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->assertValidFrom($from);
+
+        $this->authToken = $authToken;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('turbosms://%s?from=%s', $this->getEndpoint(), $this->from);
```

Comment:
```suggestion
        return sprintf('turbosms://%s?from=%s', $this->getEndpoint(), urlencode($this->from));
```

can you please add a testphase with a `from` -> `"Symfony Community"` (with a whitespace)? Thanks

Reply:
done